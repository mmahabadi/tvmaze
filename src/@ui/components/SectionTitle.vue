<script setup lang="ts">
defineProps<{ title: string }>()
</script>

<template>
  <div class="my-9">
    <div class="title-section">
      <div class="icon"><slot></slot></div>
      <h2 class="title">{{ title }}</h2>
    </div>
  </div>
</template>
<style scoped>
.title-section {
  @apply w-full flex sm:gap-8 gap-4 items-center;

  .icon {
    color: var(--primary-color);
  }
  .title {
    @apply sm:text-xl font-bold text-lg;
  }
}
</style>
