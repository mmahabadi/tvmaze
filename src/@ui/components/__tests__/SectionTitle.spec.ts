import { mount, type VueWrapper } from '@vue/test-utils'
import { beforeEach, describe, it, expect } from 'vitest'
import SectionTitle from '../SectionTitle.vue'

describe('SectionTitle', () => {
  let wrapper: VueWrapper<unknown>
  const props = { title: 'Top Rated Movies' }

  beforeEach(() => {
    wrapper = mount(SectionTitle, {
      props
    })
  })
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders the correct title', () => {
    // Arrange
    const title = wrapper.find('h2')

    // Log the title
    console.log('Title:', title.text())

    // Assert
    expect(title.text()).toBe('Top Rated Movies')
  })
})
