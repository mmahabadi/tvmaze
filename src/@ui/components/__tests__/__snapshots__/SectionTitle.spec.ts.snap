// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SectionTitle > renders correctly 1`] = `
"<div data-v-ac9d2fca="" class="my-9">
  <div data-v-ac9d2fca="" class="title-section">
    <div data-v-ac9d2fca="" class="icon"></div>
    <h2 data-v-ac9d2fca="" class="title">Top Rated Movies</h2>
  </div>
</div>"
`;
